### PR TITLE
KCL-7004 Make data attributes parser customizable

### PR DESCRIPTION
**Pull request by vladislavb-kentico**
- Data attributes parser now accepts pattern as a first argument.
- Separate patterns are used to parse data attributes for edit buttons and plus buttons.
- Tests for new data attributes parser.

### CODE REVIEW

File: test-browser/utils/dataAttributes.spec.ts [hunk 1]
```diff
@@ -1,27 +1,12 @@
...
-          done();
-        },
-        true
-      );
-
-      clickTarget.click();
-    }
-
-    it('should parse data attribute from event path', (done) => {
+  describe('parseEditButtonDataAttributes', () => {
```
Comment (Neytus) [inline test-browser/utils/dataAttributes.spec.ts:8]: It seems that all unit tests check optimistic scenarios. Would it make sense to add some more, that will make sure that also some invalid cases will be handled in the way we want? For example, not found target element, missing element codename, etc.

**Review by Neytus — COMMENTED** [review]
Thank you for this great PR, I especially like how the parser logic is complex, yet still understandable. My comments are dealing with just some details.

File: test-browser/utils/dataAttributes.spec.ts [hunk 2]
```diff
@@ -31,18 +16,61 @@ describe('dataAttributes.ts', () => {
...
-        [DataAttribute.ElementCodename, 'a'],
-        [DataAttribute.ComponentId, 'a'],
-        [DataAttribute.ItemId, 'a'],
-        [DataAttribute.LanguageCodename, 'a'],
-        [DataAttribute.ProjectId, 'a'],
-      ]);
-
-      testClickEventParsedAttributes(target, expected, done);
+      const expected = {
+        projectId: 'a',
```
[the remark this answers is not in the document]
Comment (vladislavb-kentico) [reply test-browser/utils/dataAttributes.spec.ts]: There is no reason for that. I replaced all 'a', 'b' strings with unique IDs.

File: test-browser/utils/dataAttributes.spec.ts (hunk 1 above)
Comment (vladislavb-kentico) [reply test-browser/utils/dataAttributes.spec.ts:8]: Yes, we can create more tests in the future.

**Comment by milan009** [comment]
I've went over the changes, looks great to me ✔️

**Review by milan009 — APPROVED** [review]

**Pull request merged**